### ISSUE TITLE
Full filesystem snapshoting clogs up source job

### ISSUE THREAD

**Issue by MoritzFago** [zrepl/zrepl#290]
My servers filesystem got full, by making snapshots without pruning, because the coresponding pull job didn't work.
Afer I fixed the pull job, the transfers hung until I changed the `snapshotting` `type` to `manual`. After this and a restart of the "server" zrepl service it started to transfer.

Config:
```
---
global:
  logging:
    - type: "stdout"
      level:  "error"
      format: "human"
    - type: "syslog"
      level:  "debug"
      format: "logfmt"

# Add "jobs:" section(s) here for your specific use cases.
jobs:
- name: SERVER Full
  # expect remote to connect via ssh+stdinserver with fullbackup_prod1 as client_identity
  type: source
  serve:
    type: stdinserver # see global.serve.stdinserver for explanation
    client_identities:
    - server.mydomain.foo

  filesystems: {
    "SERVER<": true,
    "SERVER/iocage/download": false,
    "SERVER/iocage/releases": false,
    "SERVER/ROOT/default/var/tmp": false,
    "SERVER/tmp":false,
  }

  snapshotting:
    type: manual
#    type: periodic
#    interval: 10m
#    interval: 1440m
#    prefix: zrepl_
```

Maybe this could be automatic?
```
sudo zrepl version
client: zrepl version= GOOS=freebsd GOARCH=amd64 Compiler=gc
server: zrepl version= GOOS=freebsd GOARCH=amd64 Compiler=gc
```
This doesn't seem that helpful. The Sha1 of the binary is:
`bdc038c46bb9da57e701b2e0ca311bfc4d5d5d25  /usr/local/bin/zrepl`
12.1-RELEASE-p2

**Comment by problame** [comment]
Please provide `pkg info zrepl` output to determine the zrepl version.

My *guess* is that the `snapshotter` required a lot of I/O and slowed down replication.
Did the transfers just hang or did they fail with an error?
Can you provide logs of the source (and pull job, if possible?)

---

@woodsb02 please make sure the FreeBSD port includes the port version in the `zrepl version` output!

**Comment by woodsb02** [comment]
> @woodsb02 please make sure the FreeBSD port includes the port version in the `zrepl version` output!

Hi Christian - any idea of the change required to make this happen? Seems like it is a change to the zrepl code?

**Comment by problame** [comment]
No, please have a look at the `Makefile`, we use `git describe` to extract the version into a variable

https://github.com/zrepl/zrepl/blob/44bd354eaedafd7eb5e92e4cf8f7b896e996459a/Makefile#L11

 and then pass it to the Go compiler's ld flags here:

https://github.com/zrepl/zrepl/blob/44bd354eaedafd7eb5e92e4cf8f7b896e996459a/Makefile#L24-L27

**Comment by problame** [comment]
(Since FreeBSD ports don't have the git history available, I suggest you override the version with whatever `pkg info` produces. If the port used our Makefile (I know it doesn't), it would just have to `make zrepl-bin ZREPL_VERSION=zrepl-$version-freebsd-$portsversionAndPossiblyGoVersion` )
